unstructured_ingest/v2/processes/connectors/discord.py

```diff
@@ -0,0 +1,148 @@
+import datetime as dt
+import logging
+from dataclasses import dataclass
+from typing import Any, Generator, Optional
+
+from pydantic import Field, Secret
+
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    Downloader,
+    DownloaderConfig,
+    DownloadResponse,
+    FileData,
+    FileDataSourceMetadata,
+    Indexer,
+    IndexerConfig,
+    SourceIdentifiers,
+)
+from unstructured_ingest.v2.processes.connector_registry import SourceRegistryEntry
+
+CONNECTOR_TYPE = "discord"
+
+
+class DiscordAccessConfig(AccessConfig):
+    token: str = Field(description="Discord API token")
+
+
+class DiscordConnectionConfig(ConnectionConfig):
+    access_config: Secret[DiscordAccessConfig] = Field(
+        default=DiscordAccessConfig, validate_default=True
+    )
+    channels: Optional[list[str]] = Field(
+        default=None, description="List of Discord channel IDs to process"
+    )
+
+    @requires_dependencies(["discord"], extras="discord")
+    def get_client(self):
```

Comment:
Can this have a response type:
```suggestion
if TYPE_CHECKING:
    from discord import Client as DiscordClient
    ...

    def get_client(self) -> "DiscordClient":
```